internal/pkg/keeper/api/caller.go

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright (C) 2024 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package api
+
+import (
+	"errors"
+
+	"github.com/edgexfoundry/go-mod-configuration/v3/internal/pkg/keeper/utils/http"
+)
+
+type Caller struct {
+	baseUrl string
+}
+
+// NewCaller creates an instance of Caller
+func NewCaller(baseUrl string) *Caller {
+	return &Caller{
+		baseUrl: baseUrl,
+	}
+}
+
+func (c *Caller) Ping() error {
+	errResp, err := http.GetRequest(nil, c.baseUrl, ApiPingRoute, nil)
+	if err != nil {
+		return err
+	}
+	if errResp.StatusCode != 0 {
+		return errors.New(errResp.Message)
+	}
+	return nil
+}
```

Comment:
IMO, the KV APIs should also be part of the object. The implementations can stay in kv.go, but change to use (ka *KeeperApi)

If you still feel the KY api needs to be separated out, then the `keeperApi` should have a reference to it via KV(), not the way you are current doing it in kv.go

Reply:
File removed, using the http clients from core-contract.